decentraland/unity-renderer
# feat: Add request events subscriptions and event updates to server

## What does this PR change?

Closes decentraland/dservices#69
Closes decentraland/dservices#70

A friendship event is any action that can have an impact on a friendship, the possible actions are:
Request, Cancel, Accepting, Rejecting, and Deleting a friendship.

This PR introduces both the listener for subscriptions for friendship events and the possibility of sending an update to the server.
This PR gives full support for the logic of the friends controller when the social service feature flag is turned on.

## How to test the changes?



1. Launch the explorer web & desktop. Its important to validate both platforms
2. Add the param to the url `&ENABLE_use-social-client`
3. Open the friends menu from the taskbar
4. You should be able to see all your friends listed
5. Make several operations with them, like removing, checking profiles, etc
6. Go to friend requests tab
7. You should see all your friend requests listed
8. Make several operations with them, like accepting, cancelling, rejecting, etc
9. Use the chat: #nearby #channels and **direct messages**
10. Pay attention on the notifications panel, it should notify you about new incoming friend requests, accepted friend requests, and the chat
11. Remove `&ENABLE_use-social-client` param
12. Check that the friends & chat feature are working as expected

This is the issue list raised from internal test sessions, which you may consider to re-validate: https://docs.google.com/spreadsheets/d/16IAV29Xiinen-UrYQo4dFzKVwgB4CvK6cyfdHIslEOg/edit#gid=0

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
